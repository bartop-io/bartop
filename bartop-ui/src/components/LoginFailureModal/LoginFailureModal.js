import React from 'react';
import PropTypes from 'prop-types';

import Modal from '../Modal/Modal';
import strings from '../../strings';

const LoginFailureModal = ({ error, ...rest }) => (
  <Modal {...rest}>
    <p>{`${strings.auth.failure} ${error ? error.message : ''}`}</p>
  </Modal>
);

LoginFailureModal.propTypes = {
  error: PropTypes.object
};

export default LoginFailureModal;
