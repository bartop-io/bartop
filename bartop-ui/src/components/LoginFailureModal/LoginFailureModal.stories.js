import React from 'react';
import { storiesOf } from '@storybook/react';

import { LoginFailureModal } from './LoginFailureModal';
import { noop } from '../../test-helpers/utils';

const props = {
  showModal: noop,
  hideModal: noop,
  ariaHideApp: false
}

storiesOf('LoginFailureModal', module).add('default', () => (
  <LoginFailureModal {...props} />
));
