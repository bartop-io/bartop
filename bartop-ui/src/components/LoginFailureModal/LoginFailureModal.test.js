import React from 'react';
import { shallow } from 'enzyme';

import { LoginFailureModal } from './LoginFailureModal';

it('matches the snapshot', () => {
  const wrapper = shallow(<LoginFailureModal />);
  expect(wrapper).toMatchSnapshot();
});
