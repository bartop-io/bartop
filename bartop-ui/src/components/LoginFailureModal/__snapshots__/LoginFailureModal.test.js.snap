// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<Styled(Modal)>
  <p>
    Authentication failed. 
  </p>
</Styled(Modal)>
`;
